File: src/main/java/com/vaadin/demo/component/icons/IconBasic.java
package com.vaadin.demo.component.icons;

import com.vaadin.flow.router.Route;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("icon-basic")
public class IconBasic extends Div {

    public IconBasic() {
        HorizontalLayout layout = new HorizontalLayout();
        layout.setSpacing(true);
        layout.setAlignItems(FlexComponent.Alignment.CENTER);

        // tag::snippet[]
        Icon pictureIcon = VaadinIcon.PICTURE.create();
        Icon phoneIcon = VaadinIcon.PHONE.create();

        layout.add(pictureIcon, phoneIcon);
        // end::snippet[]
        add(layout);
    }

    public static class Exporter extends DemoExporter<IconBasic> { // hidden-source-line
    } // hidden-source-line
}

File: src/main/frontend/demo/component/icons/icon-basic.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import '@vaadin/icons';
import '@vaadin/vaadin-lumo-styles/vaadin-iconset.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('icon-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" class="items-center">
        <!-- tag::snippet[] -->
        <vaadin-icon icon="lumo:photo"></vaadin-icon>
        <vaadin-icon icon="vaadin:phone"></vaadin-icon>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}
